packages/flutter/lib/src/widgets/notification_listener.dart

```diff
@@ -108,21 +108,7 @@ class NotificationListener<T extends Notification> extends StatelessWidget {
   bool _dispatch(Notification notification, Element element) {
     if (onNotification != null && notification is T) {
       final bool result = onNotification(notification);
-      assert(() {
-        if (result == null)
-          throw new FlutterError(
-            'NotificationListener<$T> handler returned null.\n'
-            'The onNotification handler for the NotificationListener with the '
-            'following element returned null:\n'
-            '  $element\n'
-            'The ancestor chain for this widget was as follows:\n'
-            '  ${element.debugGetCreatorChain(12)}\n'
-            'Notification listeners must return true to stop the notification bubbling, '
-            'or false to allow it to continue (the common case is returning false).'
-          );
-        return true;
-      });
-      return result;
+      return result != null && result;
```

Comment:
or rather, `return result == true; // so that null and false have the same effect` :-)